### DISCUSSION TITLE
Is it possible to preserve annotation aliases when using AnnotationToAttributeRector?

### DISCUSSION THREAD

**Discussion by jelovac** [rectorphp/rector#6660] [Q&A]
As the title says when using the AnnotationToAttributeRector rule is it possible to preserve annotation aliases?

For example I have:

```php
<?php

declare(strict_types=1);

use Doctrine\ORM\Mapping as ORM;

final class Test {

    /**
     * @ORM\Column(name="name", type="string", length=255)
     */
    protected string $name = '';

}
```

When running rector using the following configuration:

```php
    $services->set(AnnotationToAttributeRector::class)
        ->call('configure', [[
            AnnotationToAttributeRector::ANNOTATION_TO_ATTRIBUTE => ValueObjectInliner::inline([
                new AnnotationToAttribute(\Doctrine\ORM\Mapping\Column::class, null),
            ]),
        ]]);
```

I get the following:

```php
<?php

declare(strict_types=1);

use Doctrine\ORM\Mapping as ORM;

final class Test {

     #[\Doctrine\ORM\Mapping\Column(name: 'name', type: 'string', length: 255)]
    protected string $name = '';

}
```

And I want instead `#[ORM\Column(name: 'name', type: 'string', length: 255)]`

**Comment by TomasVotruba** [comment]
Not by design, it also does not make sense with attributes.

See answer in duplicated issue https://github.com/rectorphp/rector/issues/6565#issuecomment-877688006

**Reply by jelovac** [reply]
I understand that it is missing feature by original design, however I do not understand why it doesn't make sense to you that we shouldn't use import aliases with attributes.

If you take a look a the original RFC even the author proposes the use of aliases in one of the [examples](https://wiki.php.net/rfc/attributes_v2#userland_use-casemigrating_doctrine_annotations_from_docblocks_to_attributes). Searching over the Internet I couldn't find any reference which would indicate that using import aliases in annotations or attributes is a bad behavior.

**Reply by TomasVotruba** [reply]
If we forget the reason is only the past approach, why would you use aliases in your code?

**Reply by jelovac** [reply]
Without giving much thought probably because we cannot import two different classes with the same name from different namespaces. We would either need to use fully qualified classs name or give an alias to one of them or both of them. Many libraries have a common set of class names and that is why I prefer to use an alias in order to easily distinguish them.

**Reply by TomasVotruba** [reply]
That makes sense 👍
